src/main/java/liquibase/ext/opensearch/changelog/OpenSearchHistoryService.java

```diff
@@ -0,0 +1,223 @@
+package liquibase.ext.opensearch.changelog;
+
+import liquibase.ChecksumVersion;
+import liquibase.Scope;
+import liquibase.change.CheckSum;
+import liquibase.changelog.ChangeSet;
+import liquibase.changelog.RanChangeSet;
+import liquibase.database.Database;
+import liquibase.exception.DatabaseException;
+import liquibase.ext.opensearch.database.OpenSearchConnection;
+import liquibase.ext.opensearch.database.OpenSearchLiquibaseDatabase;
+import liquibase.logging.Logger;
+import liquibase.nosql.changelog.AbstractNoSqlHistoryService;
+import lombok.AllArgsConstructor;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.opensearch._types.FieldValue;
+import org.opensearch.client.opensearch._types.mapping.*;
+import org.opensearch.client.opensearch.core.SearchRequest;
+import org.opensearch.client.opensearch.core.search.Hit;
+import org.opensearch.client.opensearch.indices.PutMappingRequest;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class OpenSearchHistoryService extends AbstractNoSqlHistoryService<OpenSearchLiquibaseDatabase> {
+
+    private final Logger log = Scope.getCurrentScope().getLog(getClass());
+
+    @Override
+    protected Logger getLogger() {
+        return log;
+    }
+
+    private OpenSearchClient getOpenSearchClient() {
+        final var connection = (OpenSearchConnection) this.getNoSqlDatabase().getConnection();
+        return connection.getOpenSearchClient();
+    }
+
+    @Override
+    protected boolean existsRepository() throws DatabaseException {
+        try {
+            return this.getOpenSearchClient().indices().exists(r -> r.index(this.getDatabaseChangeLogTableName())).value();
+        } catch (final IOException e) {
+            throw new DatabaseException(e);
+        }
+    }
+
+    @Override
+    protected void createRepository() throws DatabaseException {
+        // note: the mapping will be created in adjustRepository
+
+        try {
+            this.getOpenSearchClient().indices().create(r -> r.index(this.getDatabaseChangeLogTableName()));
+        } catch (final IOException e) {
+            throw new DatabaseException(e);
+        }
+    }
+
+    @Override
+    protected void adjustRepository() throws DatabaseException {
+        // properties must match RanChangeSet
```

Comment:
it might make sense to add a test which purely tests (using reflection) that this class contains the expected list of fields. this way it'll fail automatically when we update to a newer `liquibase-core` version which brings in changes to the class. as these updates are done automatically by dependabot there's a risk that it'll bring a new version and we won't notice the changes.

@filipelautert: WDYT?